pkg/apis/monitoring/v1alpha1/prometheus_operator_types.go

```diff
@@ -26,3 +27,37 @@ type PrometheusOperatorSpec struct {
 	// Image of config-reloader sidecar container
 	ConfigmapReloaderImage string `json:"configmapReloaderImage,omitempty"`
 }
+
+type PrometheusProvisioning struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec PrometheusProvisioningSpec `json:"spec"`
+}
+
+// PrometheusProvisioningSpec extends PrometheusSpec from prometheus-operator by adding fields necessary for deployment
+type PrometheusProvisioningSpec struct {
+	prom.PrometheusSpec `json:",inline"`
+	// Namespace to deploy prometheus statefulset in.
+	Namespace string `json:"namespace,omitempty"`
+	// Namespaces to allowed to be watched by prometheus
+	// +optional
+	// kubebuilder:default=["default","kube-system"] // TODO: how can we include a field value that needs to be defined by user? // FIXME(paulfantom): controller-gen doesn't like defaulting arrays
+	Namespaces []string `json:"namespaces,omitempty"`
+}
+
+type AlertmanagerProvisioning struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec AlertmanagerProvisioningSpec `json:"spec"`
+}
+
+type AlertmanagerProvisioningSpec struct {
+	prom.AlertmanagerSpec `json:",inline"`
+	// Namespace to deploy alertmanager statefulset in.
+	Namespace string `json:"namespace,omitempty"`
+	// Embedded alertmanager configuration file. Spec as in https://prometheus.io/docs/alerting/latest/configuration/#configuration-file
+	// +optional
+	Config map[string]string `json:"config,omitempty"`
```

Comment:
What if we do not pass the whole config? Alertmanager CR already have `configSecret` and we can force users to configure their own Secret as they would need to do this either way. This would also allow easier integrations with things like SealedSecrets. Plus we wouldn't need to extend alertmanager CRD. @brancz WDYT?

Reply:
Actually not sure what we wanted to do here in the first place, so sgtm!

Reply:
Just for clarity - we wanted to incorporate this: https://github.com/prometheus-operator/kube-prometheus/blob/main/jsonnet/kube-prometheus/components/alertmanager.libsonnet#L22-L59 ;)